src/main/java/com/musala/sdcs/device/DeviceImpl.java

```diff
@@ -99,15 +112,21 @@ public void setLabel(String label) {
         this.label = label;
     }
 
+    @Override
     public List<Channel> getChannels() {
         return channels;
     }
 
+    @Override
     public void setChannels(List<Channel> channels) {
         this.channels = channels;
     }
     
     public void addChannel(Channel channel) {
```

Comment:
Will it ever happen to call this method before `this.channels` is initialized? Why don't you just initialize the instance variable
```java 
private List<Channel> channels = new ArrayList<Channel>();
```
and change the method logic to:
```java
public void addChannel(Channel channel) {
        if (this.channels != null) {
            this.channels.add(channel);
        }
    }
```
?